src/components/ScriptExplorer/IndirectSignatureImporter.jsx

```diff
@@ -0,0 +1,256 @@
+import React from "react";
+import PropTypes from "prop-types";
+import {
+  PENDING,
+  UNSUPPORTED,
+  SignMultisigTransaction,
+  ACTIVE,
+} from "unchained-wallets";
+import {
+  Box,
+  FormHelperText,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+  FormGroup,
+} from "@material-ui/core";
+import { satoshisToBitcoins } from "unchained-bitcoin";
+import InteractionMessages from "../InteractionMessages";
+
+class IndirectSignatureImporter extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      bip32PathError: "",
+      signatureError: "",
+      status: this.interaction().isSupported() ? PENDING : UNSUPPORTED,
+    };
+  }
+
+  interaction = () => {
+    const { signatureImporter, network, inputs, outputs } = this.props;
+    const keystore = signatureImporter.method;
+    const bip32Paths = inputs.map((input) => {
+      if (typeof input.bip32Path === "undefined")
+        return signatureImporter.bip32Path; // pubkey path
+      return `${signatureImporter.bip32Path}${input.bip32Path.slice(1)}`; // xpub/pubkey slice away the m, keep /
+    });
+
+    return SignMultisigTransaction({
+      keystore,
+      network,
+      inputs,
+      outputs,
+      bip32Paths,
+    });
+  };
+
+  renderDeviceConfirmInfo = () => {
```

Comment:
These types of methods should be their own component. Even if it's in the same file. Can be done in another PR of course, but just an FYI.

Reply:
Yes to this^ and it's duplicated functionality across indirect/direct rn